swagger/src/test/scala/org/http4s/rho/swagger/TypeBuilderSpec.scala

```diff
@@ -403,6 +411,18 @@ class TypeBuilderSpec extends Specification {
       modelOf[FooEither] must not(throwA[StackOverflowError])
       modelOf[FooEither].size must_== 1
     }
+
+    "Build a model for a class using type parameters of an outer class" in {
+      val ms = modelOf[OuterString.Inner]
+      ms.size must_!== 0 // At least try to return some model.
+      // It won't be a fully correct model since TypeBuilder will fail to resolve the type param T to a concrete type.
+    }
+
+    "Build a model for shapless coproduct (:+:)" in {
+      val ms = modelOf[ShapelessIntOrString]
```

Comment:
I printed the model for this one and it was.... not great.  Its awesome though that it doesn't explode.  I'll open an issue to add first class shapeless `Coproduct` swagger support.

Reply:
Issue #283 to correct the model.